Makefile

```diff
@@ -783,20 +879,23 @@ $(OBJ)/control.tar.gz: iOS/deb-postinst iOS/deb-prerm iOS/deb-control
 $(OBJ)/debian-binary:
 	-@$(MKDIR) -p $(dir $@)
 	echo 2.0 > $@
-    
-$(LIBDIR)/libsameboy.o: $(CORE_OBJECTS)
+
+# Link library objects with libtool.
+$(LIBTOOL_LIBRARY): $(CORE_LOBJECTS)
 	-@$(MKDIR) -p $(dir $@)
-	@# This is a somewhat simple hack to force Clang and GCC to build a native object file out of one or many LTO objects
-	echo "static const char __attribute__((used)) x=0;"| $(CC) $(filter-out -flto,$(CFLAGS)) -c -x c - -o $(OBJ)/lto_hack.o
-	@# And this is a somewhat complicated hack to invoke the correct LTO-enabled LD command in a mostly cross-platform nature
-	$(CC) $(FAT_FLAGS) $(CFLAGS) $(LIBFLAGS) $^ $(OBJ)/lto_hack.o -o $@
-	-@rm $(OBJ)/lto_hack.o
-    
-$(LIBDIR)/libsameboy.a: $(LIBDIR)/libsameboy.o
+	$(LIBTOOL_LD) -rpath $(libdir) $^ -o $@
```

Comment:
rpaths are a general cause for a problems and are discouraged wherever possible.

Fedora outright bans a lot of rpaths it considers "broken"
https://fedoraproject.org/wiki/Changes/Broken_RPATH_will_fail_rpmbuild#Definition_of_a_broken_RPATH

Reply:
Fwiw GNU libtool breaks without the `-rpath` (While Slibtool doesn't have this problem). Another reason to avoid using libtool here...